### PR TITLE
Splited the  install-tools into test/build groups

### PR DESCRIPTION
**Pull request by MeenuyD**
Refactor CI installation process: split install-tools into mockery/testing/build sub-targets, divide install-ci into install-test-ci and install-build-ci and update workflows accordingly.
#4846

### CODE REVIEW

**Comment by MeenuyD** [comment]
Hello @yurishkuro should I change the all ci workflow file if yes than what should be added there

File: Makefile [hunk 1]
```diff
@@ -400,6 +400,21 @@ install-tools:
 	$(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.52.1
```
Comment (yurishkuro) [inline Makefile:400]: `install-tools` should depend on new targets, not repeat the install commands

File: Makefile [hunk 2]
```diff
@@ -400,6 +400,21 @@ install-tools:
...
 	$(GO) install mvdan.cc/gofumpt@latest
```
Comment (yurishkuro) [inline Makefile:401]: your fork is stale, missing goversioninfo

File: Makefile [hunk 3]
```diff
@@ -400,6 +400,21 @@ install-tools:
...
 
+.PHONY: install-test-tools
+install-test-tools:
+    $(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.52.1
+
+.PHONY: install-build-tools
+install-build-tools:
+    $(GO) install mvdan.cc/gofumpt@latest
```
Comment (yurishkuro) [inline Makefile:409]: fumpt runs as linter, so needs to be in test-tools

build-tools would be `goversioninfo`

File: .github/workflows/ci-build-binaries.yml [hunk 4]
```diff
@@ -61,7 +61,7 @@ jobs:
       uses: ./.github/actions/setup-branch
 
     - name: Install tools
-      run: make install-ci
+      run: make install-build-ci
```
Comment (yurishkuro) [inline .github/workflows/ci-build-binaries.yml:64]: what about all other workflows?
```
$ grep -nr 'make install-ci' .github/workflows
.github/workflows/ci-release.yml:62:      run: make install-ci
.github/workflows/ci-unit-tests-go-tip.yml:34:      run: make install-ci
.github/workflows/ci-unit-tests.yml:40:      run: make install-ci
.github/workflows/ci-docker-build.yml:47:      run: make install-ci
.github/workflows/ci-build-binaries.yml:64:      run: make install-ci
.github/workflows/ci-crossdock.yml:44:      run: make install-ci
.github/workflows/ci-elasticsearch.yml:53:      run: make install-ci
.github/workflows/ci-hotrod.yml:43:      run: make install-ci
.github/workflows/ci-all-in-one-build.yml:46:      run: make install-ci
.github/workflows/ci-opensearch.yml:50:      run: make install-ci
```

Comment (MeenuyD) [reply .github/workflows/ci-build-binaries.yml:64]: In other workflow file what needs to added build,test can you tell me please

Comment (yurishkuro) [reply .github/workflows/ci-build-binaries.yml:64]: only test workflows need the test tools, everything else only build. The CI will fail if we're wrong about that so we'll find out

Comment (MeenuyD) [reply .github/workflows/ci-build-binaries.yml:64]: Hello @yurishkuro can you please look the #4859

**Pull request closed without merging**